configs/components/puppet.rb

```diff
@@ -59,6 +63,83 @@
     pkg.install_service "ext/solaris/smf/puppet.xml", "ext/solaris/smf/puppet", service_type: "network"
   when "aix"
     pkg.install_service "resources/aix/puppet.service", nil, "puppet"
+  when "windows"
+    description = "Periodically fetches and applies configurations from a Puppet master server."
+    service_hash = [
```

Comment:
I will reiterate that I think using Ruby hashes to templatize Wix XML (in this case the `ServiceInstall` element) is a problematic idea:

* The abstraction is leaky - note that XMLisms have surfaced in the `:condition` as `CDATA`, and elsewhere as entity escapes like `&quot;`.  If the goal was to insulate from the Wix details, I don't think this approach succeeds.  Now consumers must be concerned with understanding the rules of this abstraction in addition to those of Wix / XML.  I would rather see these details remain in their native Wix XML files as its a clearer demarcation.
* The XML template built at  https://github.com/puppetlabs/vanagon/blob/d03a46ededb45d3df20566b35da2b18e149cb21b/resources/windows/wix/service.component.wxs.erb has a number of hardcoded elements / doesn't fully capture the flexibility of the underlying `ServiceInstall` type at http://wixtoolset.org/documentation/manual/v3/xsd/wix/serviceinstall.html.  Specifically, `Account`, `Password`, `Interactive`, `Type`, `ErrorControl` and `Vital` are all hard-coded.  `EraseDescription`, `LoadOrderGroup` are missing, as is the support for any of the child elements.  I'm concerned now that instead of having a solution that's flexible, there's a tight coupling to enable new functionality, blocked on the release of Vanagon gem updates.  That is a major concern IMHO.
* Our new abstraction doesn't come with the same documentation as the underlying abstraction - I'd rather be able to point directly at the Wix docs (which are quite good), rather than figuring out if an issue has been encountered in this higher level abstraction (such as properly specifying a value for interpolation, etc)
* I'm concerned that this makes manual testing more difficult - i.e. I'd rather be able to simply assemble a static set of Wix artifacts to manually test package changes than require bits to be codegen'd and glued together.
* We know we need to have Wix XML in this repo specific to the project, so there's no real savings here in terms of freeing ourselves from the Wix burden.  I think it ends up making things a little more confusing to have some bits specified like this, and other bits specified elsewhere, etc.  The assembly process is more difficult to cognitively understand

I do understand that we would like to make the pieces more modular / easier to compose - but can we perhaps do that by including / assembling plain old Wix snippets?  Or some other alternative solution?  Is my understanding correct that this implementation is partially driven by the need to leverage the Vanagon `install_service` API?  Are there other details that I'm missing?

/cc @puppetlabs/windows